src/app/data-stories/[id]/page.tsx

```diff
@@ -0,0 +1,32 @@
+import HubTemplate from "@/templates/HubTemplate";
+import { getContent } from "@/utils/functions";
+import { notFound } from "next/navigation";
+
+export default async function DataStories({
+  params,
+}: {
+  params: { id: string };
+}) {
+  const { dataStories } = await getContent(["dataStories"]);
+  const selectedStorie = dataStories.find((storie: any) => {
+    const fullPath = storie.fields.path;
+    const storyId = fullPath.split("/").pop();
+
+    return storyId === params.id;
+  });
+
+  if (!selectedStorie) {
+    notFound();
+  }
```

Comment:
Instead of filtering the storyMap when we get it, what about you give an id filter to the getContent function? Maybe refactoring the `getPosts` function to accept this "dataStories" type and then use it to retrieve just this specific story map could save some network resources